_install-and-configure/upgrade-opensearch/index.md

```diff
@@ -26,41 +26,57 @@ Take time to plan the process before making any changes to your cluster. For exa
 
-### Review the OpenSearch tools compatibility matrices
+### Reviewing the OpenSearch tools compatibility matrices
 
 If your OpenSearch cluster interacts with other services in your environment, like Logstash or Beats, then you should check the [OpenSearch tools compatibility matrices]({{site.url}}{{site.baseurl}}/tools/index/#compatibility-matrices) to determine whether other components will need to be upgraded.
 
-### Review plugin compatibility
+### Reviewing plugin compatibility
 
 Review the plugins you use to determine compatibility with the target version of OpenSearch. Official OpenSearch Project plugins can be found in the [OpenSearch Project](https://github.com/opensearch-project) repository on GitHub. If you use any third-party plugins, then you should check the documentation for those plugins to determine whether they are compatible.
 
-### Back up configuration files
+Visit [Available plugins]({{site.url}}{{site.baseurl}}/install-and-configure/plugins/#available-plugins) to see a reference table that highlights version compatibility for bundled OpenSearch plugins.
 
-Mitigate the risk of data loss by backing up any important files before you start an upgrade. Generally speaking, these files will be located in either of two directories:
+Major, minor, and patch plugin versions must match OpenSearch major, minor, and patch versions in order to be compatible. For example, plugin versions 2.3.0.x work only with OpenSearch 2.3.0.
+{: .important}
+
+### Backing up configuration files
+
+Mitigate the risk of data loss by backing up any important files before you start an upgrade. Generally, these files will be located in either of two directories:
 
 - `opensearch/config`
 - `opensearch-dashboards/config`
 
-Some examples include `opensearch.yml`, `opensearch_dashboards.yml`, plugin configuration files, and TLS certificates. Once you identify which files you want to back up, copy them to remote storage for safety.
+Some examples include `opensearch.yml`, `opensearch_dashboards.yml`, plugin configuration files, and Transport Layer Security (TLS) certificates. Once you identify which files you want to back up, copy them to remote storage for safety.
+
+If you use security features, make sure to read [A word of caution]({{site.url}}{{site.baseurl}}/security-plugin/configuration/security-admin/#a-word-of-caution) for information about backing up and restoring your security settings.
+
+### Creating a snapshot
+
+We recommend that you back up your cluster state and indexes using [snapshots]({{site.url}}{{site.baseurl}}/opensearch/snapshots/index/). Snapshots you take before the upgrade can be used as restore points if you need to roll back the cluster to its original version.
 
-### Take a snapshot
+You can further reduce the risk of data loss by storing your snapshots on external storage, such as a mounted Network File System (NFS) or a cloud storage solution like one of the following:
```

Comment:
```suggestion
You can further reduce the risk of data loss by storing your snapshots in external storage, such as a mounted Network File System (NFS), or a cloud storage solution like those listed in the following table.
```

Reply:
Confirm that my change is correct (that the sentence introduces the table).

Reply:
That works perfect!